src/NServiceBus.Core/Diagnostics/Tracing.cs

```diff
@@ -0,0 +1,18 @@
+﻿namespace NServiceBus;
+
+using Microsoft.Extensions.DependencyInjection;
+using Features;
+
+class Tracing : Feature
+{
+    public Tracing()
+    {
+        EnableByDefault(); // TODO should be explicitly enabled
+        Prerequisite(c => ActivitySources.Main.HasListeners(), "No trace listeners have been registered");
```

Comment:
This does mean that the listeners MUST be configured before the NServiceBus endpoint is created or traces will not be emitted.

Reply:
yes, but from what I can tell this is required by the OT configuration API anyways? At least using the generic host?